.github/workflows/o1vm-mips-prover.yml

```diff
@@ -0,0 +1,48 @@
+name: O1VM MIPS Prover CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  run:
+    name: Run o1vm prover on all mips tests
+
+    runs-on: ["ubuntu-latest"]
+
+    strategy:
+      matrix:
+        rust_toolchain_version: ["1.74"]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Use shared Rust toolchain setting up steps
+        uses: ./.github/actions/toolchain-shared
+        with:
+          rust_toolchain_version: ${{ matrix.rust_toolchain_version }}
+
+      - name: Apply the Rust smart cacheing
+        uses: Swatinem/rust-cache@v2
+
+      - name: Build MIPS binaries
+        uses: ./.github/actions/build-mips
+
+      - name: Cache and generate SRS
+        uses: ./.github/actions/cache-srs
+
+      - name: Run the MIPS prover on all the test files
+        run: |
+          cargo build --release --bin pickles_o1vm
+          find o1vm/resources/programs/mips/bin -type f \
+            ! -name "*.*" \
+            ! -name "oracle*" \
+            ! -name "exit_group" \
+            ! -name "mul" | \
+            xargs -I {} env RUST_LOG=debug ./o1vm/test-mips-program-prover.sh {} ./srs/test_vesta.srs
```

Comment:
As discussed online, I would parallelise it, using an equivalent strategy than in [o1js](https://github.com/o1-labs/o1js/blob/main/.github/workflows/build-action.yml).